### PR TITLE
added feature file and step definition for enabling and disabling sync of share

### PR DESCRIPTION
**Pull request by S-Panta**
This PR add test for enabling and disabling sync of share in Personal Space

### CODE REVIEW

File: tests/behat/uploadFileToMoodle.feature [hunk 1]
```diff
@@ -43,3 +43,18 @@ Feature: upload the resource in oCIS to moodle
     And I click on "//*[@class='fp-filename-field']/p[text()='testfile.txt']" "xpath_element"
     And I click on "Select this file" "button"
     Then I should see "testfile.txt"
+  @skipOnStable
```
Comment (S-Panta) [inline tests/behat/uploadFileToMoodle.feature:46]: For stable branch, disabling sync would result in 200 status code. This was bug in master and was fixed but it look like it wasn't backported in stable. Thus, skipping this test in stable

Comment (individual-it) [reply tests/behat/uploadFileToMoodle.feature:46]: maybe better just to make the test more relaxed, here we don't really care or test what ocis exactly responds, just if it works for the user.

File: tests/behat/graph_helper.php [hunk 2]
```diff
@@ -248,10 +257,56 @@ public function send_share_invitation(string $user, string $sharetype, string $s
...
+        $itemid = $sharespaceid . '!' . $shareitemid;
+        $client = $this->get_client($user);
+        return $client->request(
+            'DELETE',
+            "/graph/v1beta1/drives/$sharespaceid/items/$itemid",
+        );
+    }
+
+    /**
+     * send request to disable sync of share
```
Comment (phil-davis) [inline tests/behat/graph_helper.php:289]:
```suggestion
     * send request to enable sync of share
```

File: tests/behat/uploadFileToMoodle.feature [hunk 3]
```diff
@@ -103,3 +103,18 @@ Feature: upload the resource in oCIS to moodle
     Then I should see "Personal"
     And I should see "Shares"
     And I should see "ProjectMoodle"
+  @skipOnStable @ocis-issue-8961
```
Comment (amrita-shrestha) [inline tests/behat/uploadFileToMoodle.feature:106]:
```suggestion

  @skipOnStable @ocis-issue-8961
```

File: tests/behat/graph_helper.php [hunk 4]
```diff
@@ -248,10 +257,56 @@ public function send_share_invitation(string $user, string $sharetype, string $s
         ];
         $itemid = $this->get_resource_id($user, $resource, $space);
         $client = $this->get_client($user);
-        return $client->request(
+        $response = $client->request(
             'POST',
             "/graph/v1beta1/drives/$spaceid/items/$itemid/invite",
             json_encode($body)
         );
+        $this->lastcreatedshare = $response;
```
Comment (individual-it) [inline tests/behat/graph_helper.php:265]: if `$this->lastcreatedshare` should only hold the id of the share, why putting the complete response into it?

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/behat/graph_helper.php [hunk 5]
```diff
@@ -248,10 +257,55 @@ public function send_share_invitation(string $user, string $sharetype, string $s
...
+    /**
+     * send request to disable sync of share
+     * @param string $user
+     *
+     * @return array
+     */
+    public function disable_share_sync(string $user): array {
+        $shareitemid = $this->lastcreatedshareid;
+        $sharespaceid = self::SHARES_SPACE_ID;
+        $itemid = $sharespaceid . '!' . $shareitemid;
```
Comment (individual-it) [inline tests/behat/graph_helper.php:279]:
```suggestion
        $itemid = [$sharespaceid](self::SHARES_SPACE_ID) . '!' . $this->lastcreatedshareid;
```

File: tests/behat/graph_helper.php [hunk 6]
```diff
@@ -248,10 +257,55 @@ public function send_share_invitation(string $user, string $sharetype, string $s
...
+            "remoteItem" => [
+                "id" => $itemid,
+            ],
+        ];
+        $sharespaceid = self::SHARES_SPACE_ID;
+        $client = $this->get_client($user);
+        return $client->request(
+            'POST',
+            "/graph/v1beta1/drives/$sharespaceid/root/children",
+            json_encode($body)
```
Comment (individual-it) [inline tests/behat/graph_helper.php:308]:
```suggestion
        $client = $this->get_client($user);
        return $client->request(
            'POST',
            "/graph/v1beta1/drives/" . self::SHARES_SPACE_ID . "/root/children",
            json_encode($body)
```

**Review by individual-it — COMMENTED** [review]
just some small comments

**Review by individual-it — APPROVED** [review]

**Pull request merged**